hazelcast/src/main/java/com/hazelcast/concurrent/lock/LockStore.java

```diff
@@ -20,33 +20,140 @@
 
 import java.util.Set;
 
+/**
+ * A container for multiple locks. Each lock is differentiated by a different {@code key}.
+ */
 public interface LockStore {
 
+    /**
+     * Lock a specific key.
+     *
+     * @param key         the key to lock
+     * @param caller      the identifier for the caller
+     * @param threadId    the identifier for the thread on the caller
+     * @param referenceId the identifier for the invocation of the caller (e.g. operation call ID)
+     * @param leaseTime   the lease duration in milliseconds
+     * @return if the lock was successfully acquired
+     */
     boolean lock(Data key, String caller, long threadId, long referenceId, long leaseTime);
 
+    /**
+     * Lock a specific key.
```

Comment:
This method marks the lock as transactional. It's the main difference between this and above plain `lock()`.